classes/decorator.rb

```diff
@@ -0,0 +1,11 @@
+require_relative './nameable'
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
```

Comment:
- [ ] Please make sure the nameable variable is readable and writable to work the decorate pattern as expected.
  - You can consider using the code `attr_accessor :nameable`.

Reply:
Thank you for the suggestion.